components/core/src/clp/clo/CommandLineArguments.cpp

```diff
@@ -56,6 +56,252 @@ CommandLineArguments::parse_arguments(int argc, char const* argv[]) {
+                return ParsingResult::InfoCommand;
+            }
+
+            throw invalid_argument("COMMAND not specified.");
+        }
+        switch (command_input) {
+            case (char)Command::Search:
+            case (char)Command::ExtractIr:
+                m_command = (Command)command_input;
+                break;
+            default:
+                throw invalid_argument(string("Unknown action '") + command_input + "'");
+        }
+
+        if (Command::Search == m_command) {
+            return parse_search_arguments(
+                    options_general,
+                    parsed_command_line_options,
+                    parsed.options,
+                    argc
+            );
+        } else if (Command::ExtractIr == m_command) {
+            return parse_ir_extraction_arguments(
+                    options_general,
+                    parsed_command_line_options,
+                    parsed.options,
+                    argc
+            );
+        }
+    } catch (exception& e) {
+        SPDLOG_ERROR("{}", e.what());
+        print_basic_usage();
+        cerr << "Try " << get_program_name() << " --help for detailed usage instructions" << endl;
+        return ParsingResult::Failure;
+    }
+
+    return ParsingResult::Success;
+}
+
+auto CommandLineArguments::parse_ir_extraction_arguments(
+        po::options_description const& options_general,
+        po::variables_map& parsed_command_line_options,
+        vector<po::option> const& options,
+        int argc
+) -> CommandLineArgumentsBase::ParsingResult {
+    // Define IR extraction options
+    po::options_description options_ir_extraction("IR extraction options");
+    options_ir_extraction
+            .add_options()("temp-output-dir", po::value<string>(&m_ir_temp_output_dir)->value_name("DIR"), "Temporary output directory for IR chunks while they're being written")(
+                    "target-size",
+                    po::value<size_t>(&m_ir_target_size)->value_name("SIZE"),
+                    "Target size (B) for each IR chunk before a new chunk is created"
+            );
+
+    // Define visible options
+    po::options_description visible_options;
+    visible_options.add(options_general);
+    visible_options.add(options_ir_extraction);
+
+    // Define hidden positional options (not shown in Boost's program options help message)
+    po::options_description hidden_positional_options;
+    string output_handler_name;
+    // clang-format off
+    hidden_positional_options.add_options()(
+            "archive-path",
+            po::value<string>(&m_archive_path)
+    )(
+            "file-split-id",
+            po::value<string>(&m_file_split_id)
+    )(
+            "output-dir",
+            po::value<string>(&m_ir_output_dir)
+    )(
+            "mongodb-uri",
+            po::value<string>(&m_ir_mongodb_uri)
+    )(
+            "mongodb-collection",
+            po::value<string>(&m_ir_mongodb_collection)
+    );
```

Comment:
tried. don't think it's much more readable since now you have a long line.
one option is to keep the current format and add hidden_positional_options to the beginning of every argument, but then that's a lot of copy paste.

I will keep the current format unless someone strongly oppose,